File: src/01-easy/the-river-ii-/solution.lua
function ds(n) local s=0; while n>0 do s=s+n%10; n=math.floor(n/10) end; return s end
local r=tonumber(io.read())
local lo=math.max(1,r-45)
local found=false
for x=lo,r-1 do if x+ds(x)==r then found=true; break end end
print(found and "YES" or "NO")

File: src/01-easy/the-river-ii-/solution.go
package main
import "fmt"
func ds(n int)int{s:=0;for n>0{s+=n%10;n/=10};return s}
func main(){
    var r int;fmt.Scan(&r)
    lo:=r-45;if lo<1{lo=1}
    for x:=lo;x<r;x++{if x+ds(x)==r{fmt.Println("YES");return}}
    fmt.Println("NO")
}
